src/Components/LifecycleChart/LifecycleChart.tsx

```diff
@@ -249,33 +249,28 @@ const LifecycleChart: React.FC<LifecycleChartProps> = ({ lifecycleData, viewFilt
     setHiddenSeries((prevHiddenSeries) => {
       const newHiddenSeries = new Set(prevHiddenSeries);
 
-      // Get the count of series that have actual data
-      const totalVisibleSeries = legendNames.filter((s) => s.datapoints.length > 0).length;
-
-      // Convert Set to Array with proper typing
-      const hiddenIndices = Array.from(prevHiddenSeries) as number[];
-      const currentlyHiddenCount = hiddenIndices.filter((index: number) => {
-        return (
-          typeof index === 'number' &&
-          index >= 0 &&
-          index < legendNames.length &&
-          legendNames[index] &&
-          legendNames[index].datapoints.length > 0
-        );
-      }).length;
+      // Get only series that have actual data
+      const seriesWithData = legendNames
+        .map((series, index) => ({ series, index }))
+        .filter(({ series }) => series.datapoints.length > 0);
 
-      const isCurrentlyHidden = newHiddenSeries.has(props.index);
+      // Count how many series with data are currently visible
+      const currentlyVisibleSeriesWithData = seriesWithData.filter(({ index }) => !prevHiddenSeries.has(index));
 
-      // If we're trying to hide the last visible series, don't allow it
-      if (!isCurrentlyHidden && currentlyHiddenCount >= totalVisibleSeries - 1) {
-        // Optionally show a message or just ignore the click
-        console.log('Cannot hide all series - at least one must remain visible');
-        return prevHiddenSeries; // Return unchanged state
+      const isCurrentlyHidden = newHiddenSeries.has(props.index);
+      const clickedSeriesHasData = legendNames[props.index]?.datapoints.length > 0;
+
+      // Only prevent hiding if:
+      // 1. The clicked series has data
+      // 2. It would be the last visible series with data
```

Comment:
```suggestion
      // - the series is not currently hidden (without this it wouldn't be possible unhide series in case the last one is showing) and
      // - the clicked series has data (if it doesn't, it can be hidden anyway) and
      // - it would be the last visible series with data (hiding last visible series results in empty chart)
```
Small improvement of the comment to better match the condition (the `!isCurrentlyHidden` description was missing). I also added a bit more description in brackets why it needs to be there based on my observation - it's up to you if you like it or not